test/e2e/util/util.go

```diff
@@ -1717,3 +1717,27 @@ func EnsureNoHCPPodsLandOnDefaultNode(t *testing.T, ctx context.Context, client
 		g.Expect(hcpNodeNames).To(ContainElement(pod.Spec.NodeName))
 	}
 }
+
+func EnsureSATokenNotMountedUnlessNecessary(t *testing.T, ctx context.Context, c crclient.Client) {
+	g := NewWithT(t)
+
+	var pods corev1.PodList
+	if err := c.List(ctx, &pods); err != nil {
+		t.Fatalf("failed to list pods: %v", err)
+	}
+
+	for _, pod := range pods.Items {
+		hasPrefix := false
+		for _, prefix := range expectedKasManagementComponents {
+			if strings.HasPrefix(pod.Name, prefix) {
+				hasPrefix = true
+				break
+			}
+		}
+		if !hasPrefix {
```

Comment:
why is this not checking Spec.AutomountServiceAccountToken?

Reply:
hmm wasnt aware of it, that makes things cleaner. Thanks

Reply:
just looking into it not many pods actually use this. I think keeping it the way it is may not be as clean as checking a spec in a pod but its a more accurate way of checking atm.

I can do some refactoring so the expected pods use spec.AutomountServiceAccountToken but think it falls outside of the scope of this Jira. WDYT @enxebre